### PR TITLE
fix(images) : shellchecking the core scripts and moving things around

### PR DESCRIPTION
**Pull request by floriansipp**
As discussed with @greut , shellchecking the scripts and moving some stuff around. 
This way all the possible initialisation scripts will be numbered 1 to N and put in docker-entrypoint.d and 2-run-app.sh was moved into the docker-entrypoint since this needs to be executed everytime anyway. 
Should return correct codes and catch error if there is any issues.

### CODE REVIEW

File: core/entrypoint/docker-entrypoint.sh [hunk 1]
```diff
@@ -40,5 +33,33 @@ else
...
+    #CMD="$APP_CMD_PREFIX; vglrun -d /dev/dri/$CARD /opt/VirtualGL/bin/glxspheres64"
+    #CMD="$APP_CMD_PREFIX; QT_DEBUG_PLUGINS=1 vglrun -d /dev/dri/$CARD $APP_CMD"
+    CMD="$APP_CMD_PREFIX; vglrun -d /dev/dri/$CARD $APP_CMD"
+    ;;
+esac
 
-exec "$@"
+runuser -l "$CHORUS_USER" -c "$CMD"
+retVal=$?
+exit $retVal
```
Comment (greut) [inline core/entrypoint/docker-entrypoint.sh:65]:
```suggestion
exec runuser -l "$CHORUS_USER" -c "$CMD"
```

File: core/entrypoint/docker-entrypoint.sh [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
+# Check if /docker-entrypoint.d/ contains any files
+if /usr/bin/find "/docker-entrypoint.d/" -mindepth 1 -maxdepth 1 -type f -print -quit 2>/dev/null | read -r _; then
     echo "$0: /docker-entrypoint.d/ is not empty, will attempt to perform configuration"
 
     echo "$0: Looking for shell scripts in /docker-entrypoint.d/"
-    find "/docker-entrypoint.d/" -follow -type f -print | sort -V | while read -r f; do
+    find "/docker-entrypoint.d/" -follow -type f -print | sort -V | while IFS= read -r f; do
         case "$f" in
-            *.envsh)
+            *.envsh|*.sh)
```
Comment (greut) [inline core/entrypoint/docker-entrypoint.sh:19]: they aren't _executed_ in the same fashion, hence you cannot group them. `. $f` vs `$f`

**Review by greut — CHANGES_REQUESTED** [review]

File: core/entrypoint/docker-entrypoint.sh (hunk 2 above)
Comment (floriansipp) [reply core/entrypoint/docker-entrypoint.sh]: Dang me, i got chatgpt'ed when i was looking at how to handle error catching in bash , will be more careful next time ...

File: core/entrypoint/docker-entrypoint.sh [hunk 3]
```diff
@@ -40,5 +42,31 @@ else
...
 
+# Add DISPLAY to APP_CMD_PREFIX
+if [ -n "$APP_CMD_PREFIX" ]; then
+    APP_CMD_PREFIX="export DISPLAY=$DISPLAY;$APP_CMD_PREFIX"
+else
+    APP_CMD_PREFIX="export DISPLAY=$DISPLAY"
+fi
+
+# Ensure CARD and APP_CMD are set
+: "${CARD:?Environment variable CARD is required but not set}"
```
Comment (greut) [inline core/entrypoint/docker-entrypoint.sh:53]: Can we defaults `CARD` to `none`?

**Review by iDmple — DISMISSED** [review]

File: core/entrypoint/docker-entrypoint.sh (hunk 3 above)
Comment (floriansipp) [reply core/entrypoint/docker-entrypoint.sh]: image

File: core/entrypoint/docker-entrypoint.sh [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
+# Ensure that variables are set
 if [ -z "$CHORUS_USER" ]; then
     echo "CHORUS_USER is not set. Defaulting to 'chorus'"
     export CHORUS_USER="chorus"
 fi
 
+if [ -z "$CARD" ]; then
+    echo "CARD is not set. Defaulting to 'none'"
+    CARD="none"
+fi
```
Comment (greut) [inline core/entrypoint/docker-entrypoint.sh:15]: it's an expensive way of doing this, but it's quite explicit, so good probably.

**Review by greut — APPROVED** [review]

File: core/entrypoint/docker-entrypoint.sh (hunk 4 above)
Comment (floriansipp) [reply core/entrypoint/docker-entrypoint.sh:15]: Yep, don't really know which way things are going to evolve with the different apps and cases to deal with. Once this solidifies a little i am all for bringing the machete out and simplifying

**Pull request merged**